samza-core/src/main/scala/org/apache/samza/diagnostics/DiagnosticsManager.java

```diff
@@ -93,12 +102,23 @@ public DiagnosticsManager(String jobName,
       String hostname,
       SystemStream diagnosticSystemStream,
       SystemProducer systemProducer,
-      Duration terminationDuration, boolean autosizingEnabled) {
+      Duration terminationDuration,
+      boolean autosizingEnabled,
+      String deploymentType,
```

Comment:
Should deploymentType and api type be enums, since we defined an enum for them above?